test/sourceMapPipeline/gulptask.js

```diff
@@ -0,0 +1,28 @@
+var gulp = require('gulp');
+var sourcemaps = require('gulp-sourcemaps');
+var uglify = require('gulp-uglify');
+var babel = require('gulp-babel');
+
+// gulp-typescript => babel => uglify with end-to-end sourcemaps
+// with a source directory (Main) that TypeScript removes from virtual filenames
+module.exports = function(newTS, lib, output, reporter) {
+	var project = newTS.createProject('test/sourceMapPipeline/tsconfig.json', {
+		typescript: lib
+	});
+
+	return project.src()
+		.pipe(sourcemaps.init())
+		.pipe(newTS(project, undefined, reporter)).js
+		// No sane person would ever pipe TypeScript output through
```

Comment:
This isn't true, as you can use Babel to transpile features that TypeScript cannot do yet, for instance generators. Can you remove that?